pageserver/src/tenant/vectored_blob_io.rs

```diff
@@ -50,6 +51,31 @@ pub struct VectoredBlobsBuf {
     pub blobs: Vec<VectoredBlob>,
 }
 
+/// Decompresses the buffer if needed.
+///
+/// Returns an owned buffer if decompression is performed, otherwise return `None`.
+pub(crate) async fn decompress(
+    buf: &[u8],
+    compression_bits: u8,
+) -> Result<Option<Bytes>, std::io::Error> {
+    let mut decompressed_vec = Vec::new();
```

Comment:
let's move this to the block where it's used.

Reply:
Disagree. We'd need to consume it anyway for returning it as owned in line 69

Reply:
What do you mean? It is only used in the lines 65...59, not returned.

Reply:
Ahhh, I thought you meant to move it into the caller so we could re-use the allocation (which doesn't make sense)

Reply:
Oki. I wish I could describe this in ed(1).. I don't know how to use it, but chatgpt suggests:

```
65
61t65
61d
```

:)

Reply:
bb07ab083e40182ae4c379d6097e9e671be5f701